[CI] StoreRecoveryTests testAddIndices failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #7118 / windows-2022_checkpart1_platform-support-windows](https://gradle-enterprise.elastic.co/s/fmnwzvesntsig)
- [elasticsearch-periodic-platform-support #7104 / windows-2022_checkpart1_platform-support-windows](https://gradle-enterprise.elastic.co/s/pf7akk5v6eozg)
- [elasticsearch-periodic-platform-support #7097 / windows-2022_checkpart1_platform-support-windows](https://gradle-enterprise.elastic.co/s/uact4pyhvo3o4)
- [elasticsearch-periodic-platform-support #7090 / windows-2022_checkpart1_platform-support-windows](https://gradle-enterprise.elastic.co/s/rxluxz6r533do)
- [elasticsearch-periodic-platform-support #7083 / windows-2022_checkpart1_platform-support-windows](https://gradle-enterprise.elastic.co/s/e4m24e5z3t2uc)
- [elasticsearch-periodic-platform-support #7076 / windows-2022_checkpart1_platform-support-windows](https://gradle-enterprise.elastic.co/s/bepe3xx3raavq)
- [elasticsearch-periodic-platform-support #7069 / windows-2022_checkpart1_platform-support-windows](https://gradle-enterprise.elastic.co/s/qqqvnou7en7fy)
- [elasticsearch-periodic-platform-support #7062 / windows-2022_checkpart1_platform-support-windows](https://gradle-enterprise.elastic.co/s/kwtzy54rzy75o)
- [elasticsearch-periodic-platform-support #7055 / windows-2022_checkpart1_platform-support-windows](https://gradle-enterprise.elastic.co/s/jpoqvskajrazm)
- [elasticsearch-periodic-platform-support #7048 / windows-2022_checkpart1_platform-support-windows](https://gradle-enterprise.elastic.co/s/2n25nkmlu4vbw)

**Reproduction Line:**
```
gradlew ":server:test" --tests "org.elasticsearch.index.shard.StoreRecoveryTests.testAddIndices" -Dtests.seed=82669B0473091E08 -Dtests.locale=bem -Dtests.timezone=America/North_Dakota/New_Salem -Druntime.java=24
```

**Applicable branches:**
9.0

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.index.shard.StoreRecoveryTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testAddIndices),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: expected:<0.0> but was:<51.0>
```

**Issue Reasons:**
- [9.0] 22 consecutive failures in step windows-2022_checkpart1_platform-support-windows
- [9.0] 22 failures in test testAddIndices (3.1% fail rate in 705 executions)
- [9.0] 22 failures in step windows-2022_checkpart1_platform-support-windows (100.0% fail rate in 22 executions)
- [9.0] 21 failures in pipeline elasticsearch-periodic-platform-support (95.5% fail rate in 22 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -261,9 +261,6 @@ tests:
 - class: org.elasticsearch.smoketest.DocsClientYamlTestSuiteIT
   method: test {yaml=reference/cat/nodes/line_361}
   issue: https://github.com/elastic/elasticsearch/issues/124103
-- class: org.elasticsearch.index.shard.StoreRecoveryTests
-  method: testAddIndices
-  issue: https://github.com/elastic/elasticsearch/issues/124104
 
 # Examples:
 #

